components/markets/market-filter/index.tsx

```diff
@@ -1,35 +1,126 @@
 import { observer } from "mobx-react";
-import { useEffect, useState } from "react";
-import { ChevronDown } from "react-feather";
-import ReactSelect from "react-select";
+import Image from "next/image";
+import { useContext, useEffect, useState } from "react";
+import { ChevronDown, ChevronUp } from "react-feather";
+import ReactSelect, {
+  components,
+  ControlProps,
+  MenuListProps,
+  OptionProps,
+} from "react-select";
 import {
   MarketFilter,
   MarketsListQuery,
   MarketsOrderBy,
 } from "lib/types/market-filter";
 import { findFilterIndex } from "lib/util/market-filter";
 import {
+  categoryImages,
+  currencyImages,
   filterTypes,
   marketCurrencyFilterOptions,
   marketsOrderByOptions,
   marketStatusFilterOptions,
   marketTagFilterOptions,
 } from "lib/constants/market-filter";
 import useMarketsUrlQuery from "lib/hooks/useMarketsUrlQuery";
-import { MarketActiveFilters } from "./active-filters";
+import MarketActiveFilters from "./MarketActiveFilters";
+import MarketFiltersContainer, {
+  MarketFiltersContext,
+} from "./MarketFiltersContainer";
+
+const Control = ({ children, ...props }: ControlProps<MarketFilter, false>) => {
+  const { setSelectedMenu, selectedMenu } = useContext(MarketFiltersContext);
+  const { menuIsOpen, selectProps } = props;
+  const Chevron = menuIsOpen ? ChevronUp : ChevronDown;
+
+  const onClick = () => {
+    if (selectedMenu === selectProps.placeholder) {
+      setSelectedMenu("None");
+    } else {
+      setSelectedMenu(selectProps.placeholder as any);
+    }
+  };
 
-const Control = ({ children, label, ...rest }) => {
-  const { innerProps } = rest;
-  const { onMouseDown } = innerProps;
   return (
-    <div
-      className="flex justify-center items-center pl-ztg-20  font-medium text-ztg-16-150 text-sky-600 h-ztg-44"
-      onMouseDown={onMouseDown}
-    >
-      {label}
-      <ChevronDown size={18} className="text-sky-600 ml-ztg-8 font-bold" />
-      {children}
-    </div>
+    <components.Control {...props}>
+      <div
+        className={
+          "flex justify-center items-center ml-[10px] font-medium text-ztg-16-150 h-[44px] " +
+          (menuIsOpen ? "text-black" : "text-sky-600")
+        }
+        onClick={onClick}
+      >
+        <span className="cursor-pointer">{selectProps.placeholder}</span>
+        <Chevron size={18} className="ml-ztg-8 font-bold cursor-pointer" />
+        {children}
+      </div>
+    </components.Control>
+  );
+};
+
+const Option = ({ children, ...props }: OptionProps<MarketFilter>) => {
+  const { data } = props;
+
+  const { activeFilters } = useContext(MarketFiltersContext);
+
+  const isActive = findFilterIndex(activeFilters, data) !== -1;
+
+  const imageUrl = (() => {
```

Comment:
I think it would be cleaner for this component to take stuff like `imageUrl` as an input so that it doesn't have to know about the context around it. Is is possible to structure it that way?

Reply:
I don't know how react-select would pass `imageUrl` to component. It is passed to `ReactSelect` on line 205 in this file, but I guess it would be possible if `imageUrl` is part of the option and `data` on line 63 would contain it.
Let me try

Reply:
Ah ok, I was thinking there might be something awkward like that